### PR TITLE
[Core][ContactStructuralMechanicsApplication][Hotfix] Avoid null reference mortar mapper

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

File: kratos/processes/simple_mortar_mapper_process.cpp [hunk 1]
```diff
@@ -29,8 +29,8 @@ SimpleMortarMapperProcess<TDim, TNumNodes, TVarType, THistOrigin, THistDestinati
     TVarType& ThisVariable,
     Parameters ThisParameters,
     LinearSolverType::Pointer pThisLinearSolver
-    ): mrOriginModelPart(*new ModelPart()),
-       mrDestinationModelPart(*new ModelPart()),
+    ): mOriginModelPart(ModelPart("OriginModelPart")),
```
Comment (philbucher) [inline kratos/processes/simple_mortar_mapper_process.cpp]: why not Kratos::make_shared?

Comment (RiccardoRossi) [reply kratos/processes/simple_mortar_mapper_process.cpp]: guys, the point is that i would like to have as a mechanism that when a model part is created, it is automatically added to the Model. to achieve this it is mandatory that we avoid "strange" construction mechanisms

Comment (RiccardoRossi) [reply kratos/processes/simple_mortar_mapper_process.cpp]: @loumalouomega  this also means that you should not allocate a new modelpart there. Can't you just deprecate this constructor and oblige users to pass it constructed from outside?

Comment (loumalouomega) [reply kratos/processes/simple_mortar_mapper_process.cpp]: Well, that can be done, but will require some changes

**Comment by loumalouomega** [comment]
@RiccardoRossi Done, now it just exists the constructor with tow model parts

**Review by RiccardoRossi — APPROVED** [review]
I am approving since it looks good to me.

@philbucher if you are using this please take note

**Pull request merged**